reflex/compiler/compiler.py

```diff
@@ -125,14 +129,17 @@ def _compile_page(
     imports = utils.compile_imports(imports)
 
     # Compile the code to render the component.
-    return templates.PAGE.render(
-        imports=imports,
-        dynamic_imports=component.get_dynamic_imports(),
-        custom_codes=component.get_custom_code(),
-        state_name=state.get_name(),
-        hooks=component.get_hooks(),
-        render=component.render(),
-    )
+    kwargs = {
+        "imports": imports,
+        "dynamic_imports": component.get_dynamic_imports(),
+        "custom_codes": component.get_custom_code(),
+        "hooks": component.get_hooks(),
+        "render": component.render(),
+    }
+    if state:
+        kwargs["state_name"] = state.get_name()
+
+    return templates.PAGE.render(**kwargs)
```

Comment:
```suggestion
    kwargs = {"state_name": state.get_name()} if state else {}

    return templates.PAGE.render(
        imports=imports,
        dynamic_imports=component.get_dynamic_imports(),
        custom_codes=component.get_custom_code(),
        hooks=component.get_hooks(),
        render=component.render(),
        **kwargs,
    )
```

Better type checking/hinting if we only pass dynamic kwargs for what is actually dynamic.